src/traces/pie/defaults.js

```diff
@@ -65,6 +65,15 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
         }
     }
 
+    var title = coerce('title');
+    if(title) {
+        var titlePosition = coerce('titleposition');
+
+        if(titlePosition === 'inhole' || titlePosition === 'outside') {
```

Comment:
You don't need this `if` - after `coerce`, we know `titlePosition` has a valid value. (but if you want to prohibit center positioning when there's no hole, this would be the place:
`if(!hole && titlePosition === 'middle center') traceOut.titlePosition = 'top center';`